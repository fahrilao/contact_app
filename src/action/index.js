const getData = () => dispatch => {
    // {
    //     "id": "93ad6070-c92b-11e8-b02f-cbfa15db428b",
    //     "firstName": "Bilbo",
    //     "lastName": "Baggins",
    //     "age": 111,
    //     "photo": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550"
    // },
    // {
    //     "id": "93ad6070-c92b-11e8-b02f-cbfa15db428c",
    //     "firstName": "Jane",
    //     "lastName": "Doe",
    //     "age": 111,
    //     "photo": "https://3.bp.blogspot.com/-eTPEHqY36LU/WQGEliEWHvI/AAAAAAAABpw/qxI7VKUGHacXlSIixNM725UvZCHh0yizQCLcB/s1600/atomix_user31.png"
    // },
    // {
    //     "id": "93ad6070-c92b-11e8-b02f-cbfa15db428d",
    //     "firstName": "Afro",
    //     "lastName": "Doe",
    //     "age": 111,
    //     "photo": "https://microhealth.com/assets/images/illustrations/learn-what-works-@2x.png"
    // }

    fetch(`https://simple-contact-crud.herokuapp.com/contact`)
        .then(res => res.json())
        .then(data => {
            dispatch({
                type: 'GET_DATA',
                datas: data.data
            })
        })
        .catch(err => console.error(err))
}

const getDetail = (contact = null) => {
    return {
        type: 'GET_DETAIL',
        data: contact
    }
}

const updateContact = (contact = null) => dispatch => {
    fetch(`https://simple-contact-crud.herokuapp.com/contact`)
        .then(res => res.json())
        .then(data => {
            dispatch({
                type: 'GET_UPDATE_DETAIL',
                datas: data.data,
                data: contact
            })
        })
        .catch(err => console.error(err))
}

export { getData, getDetail, updateContact}