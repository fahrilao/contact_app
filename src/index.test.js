import configureStore from "redux-mock-store"
import { Provider } from "react-redux"
import renderer from 'react-test-renderer'

import thunk from "redux-thunk"
import indexComponent from "."
import { updateContact } from "./action"

const mockStore = configureStore([thunk]);

describe("Contact Actions", () => {
    let store
    let component

    beforeEach(() => {
        store = mockStore({
            listContact: []
        })

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <indexComponent />
            </Provider>
        );
    })

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findById('get__detail-contact').props.onClick();
        });

        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            updateContact({ payload: 'sample text' })
        );
    });
})