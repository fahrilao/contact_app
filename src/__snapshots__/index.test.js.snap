// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Actions should render with given state from Redux store 1`] = `undefined`;
